app/models/data_request_area.rb

```diff
@@ -41,12 +42,28 @@ def kase
     offender_sar_case
   end
 
+  def completed?
+    status == :completed
+  end
+
   def status
     return :not_started unless data_requests.exists?
 
     data_requests.map(&:status).all?(:completed) ? :completed : :in_progress
   end
 
+  def recipient_emails
+    contact&.data_request_emails&.split(" ") || []
+  end
+
+  def create_commissioning_document
+    template_name = data_request_area_type == "mappa" ? "mappa" : "standard"
```

Comment:
`create_commissioning_document` and `create_commissioning_document!` methods will already exist on the object (https://guides.rubyonrails.org/association_basics.html#has-one-association-reference) to create the association. They probably do the same thing as you're doing here, but it's more idiomatic to use them.

Maybe do it in a block with after_create to avoid the naming confusion?